### ISSUE TITLE
[no-this-alias] does not detect all cases of aliasing `this` to local variable

### ISSUE THREAD

**Issue by fushi** [typescript-eslint/typescript-eslint#4690]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
   "@typescript-eslint/no-this-alias": "error",
  }
}
```

```TS
let that;

that = this;
```

**Expected Result**

3:5  error  Unexpected aliasing of 'this' to local variable  @typescript-eslint/no-this-alias

**Actual Result**

No error or warning.

**Additional Info**

https://github.com/sindresorhus/eslint-plugin-unicorn catches this with unicorn/no-this-assignment

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `15.5.0` |
| `@typescript-eslint/parser`        | `15.5.0` |
| `TypeScript`                       | `4.6.2` |
| `ESLint`                           | `8.9.0` |
| `node`                             | `16.13.2` |

**Comment by juank1809** [comment]
This is because the rule only handles the case when a variable is declared with the value of **this**, ignoring when you are assigning it to a variable already declared.

I'd be glad to submit a PR to this issue.

**Comment by JoshuaKGoldberg** [comment]
Hmm, I'm a _little_ worried that this would cause new rule reports in cases that are actually fine in the spirit of the rule. Vaguely:

```ts
class SomeClass {
  doThing(target, useSelf) {
    target.work(useSelf ? target : this);
  }
}
```

I wonder if there should be an option around this behavior?

I'll defer to @bradzacher - it seems fine to me, but just wanted to raise.

**Comment by bradzacher** [comment]
Based on the current PR #4718 - this wouldn't match on that junk.
Specifically it'll just be handling `thing = this` - the `AssignmentExpression` case.
